src/sb_handle/sycl_blas_handle.hpp

```diff
@@ -130,20 +134,40 @@ inline typename SB_Handle::event_t SB_Handle::execute(
       auto localTree = expression_tree_t(((nWG == 1) ? lhs : opShMem1), rhs,
                                          localSize, globalSize);
       event.push_back(execute_tree<using_local_memory::enabled>(
-          q_, localTree, localSize, globalSize, sharedSize));
+          q_, localTree, localSize, globalSize, sharedSize, dependencies));
     } else {
       // THE OTHER CASES ALWAYS USE THE BINARY FUNCTION
       auto localTree = AssignReduction<operator_t, lhs_t, lhs_t>(
           ((nWG == 1) ? lhs : (even ? opShMem2 : opShMem1)),
           (even ? opShMem1 : opShMem2), localSize, globalSize);
       event.push_back(execute_tree<using_local_memory::enabled>(
-          q_, localTree, localSize, globalSize, sharedSize));
+          q_, localTree, localSize, globalSize, sharedSize, event));
     }
     _N = nWG;
     nWG = (_N + (2 * localSize) - 1) / (2 * localSize);
     frst = false;
     even = !even;
   } while (_N > 1);
+
+  // Need to add this guard since the enqueue_deallocate
+  // function requires a host_task which throws a runtime
+  // exception when used with the enable_profiling{} queue
+  // property. We need to create all intermediate memory
+  // before launching the kernel to avoid running into this
+  // issue.
+  // Enabling this code only for DEFAULT_CPU backend to get the
+  // CI to pass.
```

Comment:
Would it make sense to add this to all backends? What would happen if CI happen to test a different backend (or all of them)?

Reply:
I have tested these changes with `DEFAULT_CPU` and `INTEL_GPU` backends and both of them seem to be working fine. I am going to test the `NVIDIA_GPU` and the `AMD_GPU` backends as well to make sure there is no issue there as well.